cmd/systray/systray.go

```diff
@@ -375,6 +402,466 @@ func sendNotification(title, content string) {
+
+			for _, city := range country.sortedCities() {
+				cityMenu := countryMenu.AddSubMenuItemCheckbox(city.name, "", false)
+				onClick(cityMenu, city.best.ID)
+				if status.ExitNodeStatus != nil {
+					for _, ps := range city.peers {
+						if status.ExitNodeStatus.ID == ps.ID {
+							mullvadMenu.Check()
+							countryMenu.Check()
+							cityMenu.Check()
+						}
+					}
+				}
+			}
+		}
+	}
+
+	// TODO: "Allow Local Network Access" and "Run Exit Node" menu items
+}
+
+// mullvadPeers contains all mullvad peer nodes, sorted by country and city.
+type mullvadPeers struct {
+	countries map[string]*mvCountry // country code => country
+}
+
+// sortedCountries returns countries containing mullvad nodes, sorted by name.
+func (mp mullvadPeers) sortedCountries() []*mvCountry {
+	countries := make([]*mvCountry, 0, len(mp.countries))
+	for _, country := range mp.countries {
+		countries = append(countries, country)
+	}
+	sort.Slice(countries, func(i, j int) bool {
+		return strings.ToLower(countries[i].name) < strings.ToLower(countries[j].name)
+	})
+	return countries
+}
+
+type mvCountry struct {
+	code   string
+	name   string
+	best   *ipnstate.PeerStatus // highest priority peer in the country
+	cities map[string]*mvCity   // city code => city
+}
+
+// sortedCities returns cities containing mullvad nodes, sorted by name.
+func (mc *mvCountry) sortedCities() []*mvCity {
+	cities := make([]*mvCity, 0, len(mc.cities))
+	for _, city := range mc.cities {
+		cities = append(cities, city)
+	}
+	sort.Slice(cities, func(i, j int) bool {
+		return cities[i].name < cities[j].name
+	})
+	return cities
+}
+
+type mvCity struct {
+	name  string
+	best  *ipnstate.PeerStatus // highest priority peer in the city
+	peers []*ipnstate.PeerStatus
+}
+
+func newMullvadPeers(status *ipnstate.Status) mullvadPeers {
+	countries := make(map[string]*mvCountry)
+	for _, ps := range status.Peer {
+		if !ps.ExitNodeOption || ps.Location == nil {
+			continue
+		}
+		loc := ps.Location
+		country, ok := countries[loc.CountryCode]
+		if !ok {
+			country = &mvCountry{
+				code:   loc.CountryCode,
+				name:   loc.Country,
+				cities: make(map[string]*mvCity),
+			}
+			countries[loc.CountryCode] = country
+		}
+		city, ok := countries[loc.CountryCode].cities[loc.CityCode]
+		if !ok {
+			city = &mvCity{
+				name: loc.City,
+			}
+			countries[loc.CountryCode].cities[loc.CityCode] = city
+		}
+		city.peers = append(city.peers, ps)
+		if city.best == nil || ps.Location.Priority > city.best.Location.Priority {
+			city.best = ps
+		}
+		if country.best == nil || ps.Location.Priority > country.best.Location.Priority {
+			country.best = ps
+		}
+	}
+	return mullvadPeers{countries}
+}
+
+var countryFlags = map[string]string{
+	"ad": "🇦🇩",
+	"ae": "🇦🇪",
+	"af": "🇦🇫",
+	"ag": "🇦🇬",
+	"ai": "🇦🇮",
+	"al": "🇦🇱",
+	"am": "🇦🇲",
+	"ao": "🇦🇴",
+	"aq": "🇦🇶",
+	"ar": "🇦🇷",
+	"as": "🇦🇸",
+	"at": "🇦🇹",
+	"au": "🇦🇺",
+	"aw": "🇦🇼",
+	"ax": "🇦🇽",
+	"az": "🇦🇿",
+	"ba": "🇧🇦",
+	"bb": "🇧🇧",
```

Comment:
This is all unnecessary.

You can build the emoji programmatically from the two letters: https://go.dev/play/p/rKXgwaHhRAG

Reply:
oooh!  TIL